swift/extractor/SwiftExtractor.cpp

```diff
@@ -135,12 +148,16 @@ void codeql::extractSwiftFiles(const SwiftExtractorConfiguration& config,
       // TODO: pass ModuleDecl directly when we have module extraction in place?
       extractDeclarations(config, decls, compiler, *module);
     } else {
-      // The extraction will only work if one (or more) `-primary-file` CLI option is provided,
-      // which is what always happens in case of `swift build` and `xcodebuild`
-      for (auto primaryFile : module->getPrimarySourceFiles()) {
-        archiveFile(config, *primaryFile);
-        extractDeclarations(config, primaryFile->getTopLevelDecls(), compiler, *module,
-                            primaryFile);
+      for (auto file : module->getFiles()) {
+        if (!llvm::isa<swift::SourceFile>(file)) {
+          continue;
+        }
+        auto sourceFile = llvm::cast<swift::SourceFile>(file);
+        if (sourceFiles.count(sourceFile->getFilename().str()) == 0) {
+          continue;
+        }
+        archiveFile(config, *sourceFile);
+        extractDeclarations(config, sourceFile->getTopLevelDecls(), compiler, *module, sourceFile);
```

Comment:
mainly cosmetic, but it's shorter to combine `isa` and `cast` into a single `dyn_cast` invcation:
```suggestion
        if (auto sourceFile = llvm::dyn_cast<swift::SourceFile>(file));
            sourceFile && sourceFiles.count(sourceFile->getFilename().str) {
          archiveFile(config, *sourceFile);
          extractDeclarations(config, sourceFile->getTopLevelDecls(), compiler, *module, sourceFile);
        }
```

Reply:
IMO, compound `if` check is less readable, but I don't have strong opinions here 😄